packages/playwright-core/src/server/trace/viewer/traceViewer.ts

```diff
@@ -127,7 +127,11 @@ export async function installRootRedirect(server: HttpServer, traceUrls: string[
 
   params.set('server', server.urlPrefix('precise'));
 
-  const urlPath  = `./trace/${options.webApp || 'index.html'}?${params.toString()}`;
+  let baseUrl = '.';
+  if (process.env.PW_HMR)
+    baseUrl = 'http://localhost:44223'; // port is hardcoded in build.js
```

Comment:
Don't we need `?server` parameter? Perhaps I've lost where it is added, could you please add a comment explaining?